src/rust/engine/process_execution/src/docker.rs

```diff
@@ -544,6 +597,110 @@ impl<'a> CapturedWorkdir for CommandRunner<'a> {
   }
 }
 
+async fn pipe_inputs(
+  docker: &Docker,
+  store: &Store,
+  executor: &Executor,
+  container_id: &str,
+  working_dir: &str,
+  inputs_digest: DirectoryDigest,
+) -> Result<(), ProcessError> {
+  let config = bollard::exec::CreateExecOptions {
+    cmd: Some(vec!["tar", "-xf", "-"]),
+    working_dir: Some(working_dir),
+    attach_stdin: Some(true),
+    attach_stdout: Some(true),
+    attach_stderr: Some(true),
+    ..bollard::exec::CreateExecOptions::default()
+  };
+
+  let exec = docker
+    .create_exec(container_id, config)
+    .await
+    .map_err(|err| format!("Failed to create Docker execution in container: {err:?}"))?;
+
+  let exec_result = docker
+    .start_exec(&exec.id, None)
+    .await
+    .map_err(|err| format!("Failed to start Docker execution `{}`: {:?}", &exec.id, err))?;
+  let StartExecResults::Attached { mut output, mut input } = exec_result else {
```

Comment:
This syntax is new to me. I'd expect an `if let` would be needed to support the else at the end. I feel like I'm reading scala and space can invoke a method and else is a method or something.

Reply:
Alrighty, pretty new: https://doc.rust-lang.org/nightly/rust-by-example/flow_control/let_else.html

Reply:
This is `let-else`, but used with something other than `Option`: https://rust-lang.github.io/rfcs/3137-let-else.html